src/commonmark/commonmarkdataprocessor.js

```diff
@@ -102,17 +102,22 @@ export default class CommonMarkDataProcessor {
 		turndownService.addRule('todolist', {
 			filter: function (node) {
 				// check if we're a todo list item
-				return node.nodeName === 'LI' && node.closest('ul.todo-list');
```

Comment:
basically the difference is `node.closest` was searching all the way up the tree, so in child lists it was always finding a parent. The new code searches for the closest `ul`, and then checks if it is a todo-list or not.